src/riak_core_util.erl

```diff
@@ -704,6 +712,124 @@ proxy(Parent, Fun) ->
             ok
     end.
 
+-spec enable_job_class(atom(), atom()) -> ok | {error, term()}.
+%% @doc Enables the specified Application/Operation job class.
+%% This is the public API for use via RPC.
+enable_job_class(Application, Operation)
+        when erlang:is_atom(Application) andalso erlang:is_atom(Operation) ->
+    enable_job_class({Application, Operation});
+enable_job_class(Application, Operation) ->
+    {error, {badarg, {Application, Operation}}}.
+
+-spec disable_job_class(atom(), atom()) -> ok | {error, term()}.
+%% @doc Disables the specified Application/Operation job class.
+%% This is the public API for use via RPC.
+disable_job_class(Application, Operation)
+        when erlang:is_atom(Application) andalso erlang:is_atom(Operation) ->
+    disable_job_class({Application, Operation});
+disable_job_class(Application, Operation) ->
+    {error, {badarg, {Application, Operation}}}.
+
+-spec job_class_enabled(atom(), atom()) -> boolean() | {error, term()}.
+%% @doc Reports whether the specified Application/Operation job class is enabled.
+%% This is the public API for use via RPC.
+job_class_enabled(Application, Operation)
+        when erlang:is_atom(Application) andalso erlang:is_atom(Operation) ->
+    job_class_enabled({Application, Operation});
+job_class_enabled(Application, Operation) ->
+    {error, {badarg, {Application, Operation}}}.
+
+-spec enable_job_class(term()) -> ok | {error, term()}.
+%% @doc Internal API to enable the specified job class.
+%% WARNING:
+%% * This function may not remain in this form once the Jobs API is live!
+%% * Parameter types ARE NOT validated by the same rules as the public API!
+%% You are STRONGLY advised to use enable_job_class/2.
+enable_job_class(Class) ->
```

Comment:
I'm fairly certain both enable_job_class/1 and disable_job_class/1 contain race conditions (read, check, write with no way to do a "transaction" across the environment variable).

We may have to split the environment variable out into separate entries for each Application/Operation pair at this point? Do something similar to riak_core_throttle, which would allow you to just enable/disable an operation directly (which, admittedly, may still have multiple processes writing to it, but you wouldn't end up with one setting simply not being set at all because another overwrote the list after you did).

I'm not sure if we really _need_ a _list_ of job classes somewhere in your future design, but if we really do, we may have to punt and manage our own ETS table here to fix the overwrite race condition, where the table would have one row per application/operation pair.